### PR TITLE
fix for maps without soapdm spawns

### PR DESCRIPTION
**Pull request by atagen**
per https://github.com/sapphonie/SOAP-TF2DM/issues/9 :

it seems the core problem in the above issue occurs when there are no spawns for the map, all download attempts fail, etc. so `g_bSpawnMap` is never set `true` and there is nothing to retrieve in `g_h(Blu/Red)Spawns`.
afaict this results in `RandomSpawn` never being called, so the `TFCond_UberchargedHidden` condition set in `Event_player_spawn` is never fixed.

I understand this is probably a niche case, but in situations where we are playing newer versions of maps or scrim-testing new maps entirely this makes the pregame suck.
also, ime using the previous map version's spawns is not always desirable , as significant changes may have occurred leading to being stuck in walls etc (for example, provided warmtic cfg used on latest f5 version).

the solution I have implemented is to harvest the origin and rotation of `info_player_teamspawn` ents to fill the spawn arrays and set `g_bSpawnMap` to true, so the typical code path has something to operate on and everything else is congruent with regular operation of the plugin.

please note this diversion also results in several calls to set a failed state being downgraded to log messages to allow execution to flow on to default spawn setting routine.

there is not much error handling currently as I assume any valid map will have at least one `info_player_teamspawn` for each team, so I can't think of much to fail.

current testing on a live server seems to work well so far.

please let me know if this suits your vision for the repo, or any changes you might like to see to accept pr.

thanks

### CODE REVIEW

**Comment by atagen** [comment]
now rebased on sapphonie/alpha branch, found to be working reliably in testing

File: addons/sourcemod/scripting/soap_tf2dm.sp [hunk 1]
```diff
@@ -356,22 +354,70 @@ void InitSpawnSys()
...
+}
+
+public LoadMapConfig(const char[] map, const char[] path) {
     g_bSpawnMap = true;
     FileToKeyValues(g_hKv, path);
 
     float vectors[6];
     float origin[3];
     float angles[3];
+    bool defaults[2] = {false, false};
```
Comment (sapphonie) [inline addons/sourcemod/scripting/soap_tf2dm.sp:420]: this defaults to false you don't need to explicitly set it

File: addons/sourcemod/scripting/soap_tf2dm.sp [hunk 2]
```diff
@@ -356,22 +354,70 @@ void InitSpawnSys()
...
             LogError("File Not Found: %s, no cURL support", path);
+            SetDefaultSpawns(true, true);
         }
     }
     // End spawn system.
 }
 
-// lmfao dude kill me
-public LoadMapConfig(const char[] map, const char[] path)
+public SetDefaultSpawns(bool setRed, bool setBlu)
```
Comment (sapphonie) [inline addons/sourcemod/scripting/soap_tf2dm.sp:365]: this can just be `void SetDefaultSpawns(bool setRed, bool setBlu)`, no need to make it public

**Review by sapphonie — APPROVED** [review]
lgtm!

**Pull request merged**